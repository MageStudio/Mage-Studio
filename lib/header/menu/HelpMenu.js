import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Dropdown, Menu, Divider } from 'antd';

const { SubMenu, Item } = Menu;

const NEW = 'new';
const SAVE = 'save';
const OPEN = 'open';

export class HelpMenu extends Component {

    onClick = ({ key }) => {
        switch(key) {
            case SAVE:
                break;
            default:
                break;

        }
    }

    getMenuContent = () => (
        <Menu onClick={this.onClick}>
            <Item className="option" key="save_scene">Documentation</Item>
            <Item className="option" key="save_file">Terms and conditions</Item>
            <Item className="option" key="faq">Frequently asked questions</Item>
            <Divider/>
            <Item className="option" key="report_issue">Report issue</Item>
            <Divider/>
            <Item className="option" key="welcome">Welcome</Item>
            <Item className="option" key="about">About Mage Studio</Item>
        </Menu>
    )

    render() {
        return (
            <Dropdown overlay={this.getMenuContent()}>
                <span className="main-menu-item">
                    Help
                </span>
            </Dropdown>
        );
    }
}

export default HelpMenu;
