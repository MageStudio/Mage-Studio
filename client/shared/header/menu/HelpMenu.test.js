import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import HelpMenu from './HelpMenu';

describe('HelpMenu', () => {

    it('should render fine', () => {
        const component = shallow(<HelpMenu />);
        expect(toJSON(component)).toMatchSnapshot();
    });
})
