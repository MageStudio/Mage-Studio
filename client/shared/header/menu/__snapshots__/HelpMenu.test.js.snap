// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpMenu should render fine 1`] = `
<Dropdown
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  overlay={
    <Menu>
      <MenuItem
        className="option"
      >
        Documentation
      </MenuItem>
      <MenuItem
        className="option"
      >
        Terms and conditions
      </MenuItem>
      <MenuItem
        className="option"
      >
        Frequently asked questions
      </MenuItem>
      <Divider />
      <MenuItem
        className="option"
      >
        Report issue
      </MenuItem>
      <Divider />
      <MenuItem
        className="option"
      >
        Welcome
      </MenuItem>
      <MenuItem
        className="option"
      >
        About Mage Studio
      </MenuItem>
    </Menu>
  }
>
  <span
    className="main-menu-item"
  >
    Help
  </span>
</Dropdown>
`;
